src/api/routes/offer.js

```diff
@@ -28,6 +29,25 @@ module.exports = (app) => {
         }
     });
 
+    /**
+     * Gets an offer from the database with its id
+    */
+    router.get("/:offer", validators.get, async (req, res, next) => {
```

Comment:
You are using the same middleware for generic offer search (`validators.get`), you should create another one since you only receive one query param which is the offer id

Reply:
What should be in this middleware? Something like check if it's a valid id?

Reply:
@DoStini exactly, just verifying the parameters it should receive. In this case it will only be an id which should be a mongoose object id. I think we have an example of that in a validator somewhere.